evalml/pipelines/pipeline_base_meta.py

```diff
@@ -31,3 +31,28 @@ def _check_for_fit(self, X=None, y=None, objective=None):
             else:
                 return method(self, X, y)
         return _check_for_fit
+
+class TimeSeriesPipelineBaseMeta(PipelineBaseMeta):
+    """Metaclass that overrides creating a new pipeline by wrapping methods with validators and setters"""
+
+    @classmethod
+    def check_for_fit(cls, method):
+        """`check_for_fit` wraps a method that validates if `self._is_fitted` is `True`.
+            It raises an exception if `False` and calls and returns the wrapped method if `True`.
+        """
+        @wraps(method)
+        def _check_for_fit(self, X=None, y=None, objective=None):
+            klass = type(self).__name__
```

Comment:
Hahaha, really?

Reply:
this is what I get for trying to do work mindlessly between meetings 😭